orca-deploymentmonitor/src/main/java/com/netflix/spinnaker/orca/deploymentmonitor/models/RequestBase.java

```diff
@@ -42,6 +81,7 @@ protected void fromStage(Stage stage) {
     application = stage.getExecution().getApplication();
     executionId = stage.getExecution().getId();
     stageId = stage.getId();
+    deploymentId = stage.getParentStageId();
```

Comment:
does stage id actually identify a deployment? Why not execution id?

Reply:
The `createServerGroup` (parent) stage does - `executionId` is already there but it doesn't work when the same pipeline has multiple deploy stages. `createServerGroup` works because even if a deploy stage deploys N clusters there will be N `createServerGroup` stages and all steps will be their children